common/workunit/workunit.cpp

```diff
@@ -3741,15 +3741,23 @@ IJlibDateTime & CLocalWorkUnit::getTimeScheduled(IJlibDateTime &val) const
 bool modifyAndWriteWorkUnitXML(char const * wuid, StringBuffer & buf, StringBuffer & extra, IFileIO * fileio)
 {
     // kludge in extra chunks of XML such as GraphProgress and GeneratedDlls
-    if(extra.length())
+    if (extra.length())
     {
+        const char *bufStart = buf.str();
```

Comment:
Isn't it basically a strrchr(bufStart, '<') followed by a sanity check that the < in question is followed by "/wuid" ?

Reply:
yes, that's all this code is doing, but explicitly skipping back over whitespace, and checking '>' and </wuid